### PR TITLE
Completed array-manipulation-2 Jimmy Tran

### PR DESCRIPTION
**Pull request by JimmyTran1206**
Animation

### CODE REVIEW

File: array-manipulation-2/src/chunk.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+/* exported chunk */
+const chunk = (array, size) => {
+  if (size >= array.length) {
+    return array;
+  } else {
+    // const arrCopy=[...array];
```
Comment (shawnkost) [inline array-manipulation-2/src/chunk.js:6]: Remove all old code!

File: array-manipulation-2/src/chunk.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    }
+    if (remainingNumber !== 0) {
+      const remainingArray = array.slice(-remainingNumber); // may need more loops if not using slice
+      result.push(remainingArray);
+      return result;
+    } else {
+      return result;
+    }
+  }
+};
```
Comment (shawnkost) [inline array-manipulation-2/src/chunk.js:29]: We can actually simplify this code a lot! Check out our solution
```
function chunk(array, size) {
  const chunks = [];
  for (let i = 0; i < array.length; i++) {
    if (i % size === 0) {
      chunks.push([array[i]]);
    } else {
      const lastChunk = chunks[chunks.length - 1];
      lastChunk.push(array[i]);
    }
  }
  return chunks;
}
```

**Review by shawnkost — APPROVED** [review]

File: array-manipulation-2/src/chunk.js (hunk 2 above)
Comment (JimmyTran1206) [reply array-manipulation-2/src/chunk.js:29]: Thank you Shawn, I understand my code is clumsy. I will try to keep it simple and refactor codes next time.

Comment (shawnkost) [reply array-manipulation-2/src/chunk.js:29]: No need to apologize Jimmy! I just wanted to show you another way of accomplishing this!

**Pull request merged**